### PR TITLE
[KAFKA-16435] Add test for KAFKA-16428

### PR DESCRIPTION
**Pull request by brandboat**
related to KAFKA-16435

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by mcmmining — APPROVED** [review]

File: core/src/test/scala/unit/kafka/zk/migration/ZkConfigMigrationClientTest.scala [hunk 1]
```diff
@@ -112,10 +113,25 @@ class ZkConfigMigrationClientTest extends ZkMigrationTestHarness {
...
     assertEquals(0, zkClient.getEntityConfigs(ConfigType.BROKER, "1").size())
+    assertPathExistenceAndData("/config/changes/config_change_0000000001", """{"version":2,"entity_path":"brokers/1"}""")
+
+    // make sure there is no more config change notification in znode
+    assertFalse(zkClient.pathExists("/config/changes/config_change_0000000002"))
+  }
+
+  private def assertPathExistenceAndData(expectedPath: String, data: String): Unit = {
+    assertTrue(zkClient.pathExists(expectedPath))
+    assertEquals(Some(data), dataAsString(expectedPath))
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/zk/migration/ZkConfigMigrationClientTest.scala]: how about `assertEquals(Some(data), zkClient.getDataAndStat(expectedPath)._1.map(new String(_, UTF_8)))`?

**Review by chia7712 — COMMENTED** [review]
@brandboat nice test. one trivial comment left.

File: core/src/test/scala/unit/kafka/zk/migration/ZkConfigMigrationClientTest.scala (hunk 1 above)
Comment (brandboat) [reply core/src/test/scala/unit/kafka/zk/migration/ZkConfigMigrationClientTest.scala]: @chia7712 , many thanks ! Already addressed the comment in the latest commit :smiley:

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**